dev/npm.ts

```diff
@@ -1,50 +1,44 @@
+const searchGenerator = {
+  script: function (context) {
+    if (context[context.length - 1] === "") return "";
+    const searchTerm = context[context.length - 1];
+    return `npm search ${searchTerm}`;
+  },
+  postProcess: function (out) {
+    const allLines: Array<Record<string, string>> = out
+      .split("\n")
+      .filter(
+        (line) =>
+          !line.includes(
+            '"NAME | DESCRIPTION | AUTHOR | DATE | VERSION | KEYWORDS '
+          )
+      )
+      .map((item) => item.split("|")[0].trim());
+    return allLines;
+  },
+  trigger: function () {
+    return true;
+  },
+};
+
```

Comment:
We should add the `Fig.Generator` type to prevent mistakes, so we will need to cast the return of the `postProcess` function to `Fig.Suggestion[]`:
```suggestion
const searchGenerator: Fig.Generator = {
  script: function (context) {
    if (context[context.length - 1] === "") return "";
    const searchTerm = context[context.length - 1];
    return `npm search ${searchTerm}`;
  },
  postProcess: function (out) {
    return out
      .split("\n")
      .filter(
        (line) =>
          !line.includes(
            '"NAME | DESCRIPTION | AUTHOR | DATE | VERSION | KEYWORDS '
          )
      )
      .map((item) => item.split("|")[0].trim()) as Fig.Suggestion[];
    return allLines;
  },
  trigger: function () {
    return true;
  },
};
```